// @cucumber/cucumber-expressions/dist/src/GroupBuilder.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const Group_1 = __importDefault(require("./Group"));
class GroupBuilder {
    constructor() {
        this.capturing = true;
        this.groupBuilders = [];
    }
    add(groupBuilder) {
        this.groupBuilders.push(groupBuilder);
    }
    build(match, nextGroupIndex) {
        const groupIndex = nextGroupIndex();
        const children = this.groupBuilders.map((gb) => gb.build(match, nextGroupIndex));
        return new Group_1.default(match[groupIndex] || undefined, match.index[groupIndex], match.index[groupIndex] + (match[groupIndex] || '').length, children);
    }
    setNonCapturing() {
        this.capturing = false;
    }
    get children() {
        return this.groupBuilders;
    }
    moveChildrenTo(groupBuilder) {
        this.groupBuilders.forEach((child) => groupBuilder.add(child));
    }
}
exports.default = GroupBuilder;
